build break with go 1.10

As of 172780687795638290f94acabf99246cf46f4b1f (committed by @xiang90), building with Go 1.10.8 gives:

```
# go.etcd.io/bbolt
./bolt_riscv64.go:5:20: maxMapSize redeclared in this block
	previous declaration at ./bolt_amd64.go:4:20
./bolt_riscv64.go:8:22: maxAllocSize redeclared in this block
	previous declaration at ./bolt_amd64.go:7:22
./bolt_riscv64.go:11:5: brokenUnaligned redeclared in this block
	previous declaration at ./bolt_amd64.go:10:5
FAIL	go.etcd.io/bbolt [build failed]
```

The fix is that all bolt_$ARCH.go files need a build tag at the front of them, so that Go compilers that do not know about newer architectures will not build the files:

```diff
diff --git a/bolt_riscv64.go b/bolt_riscv64.go
index 2d962d3..07b4b47 100644
--- a/bolt_riscv64.go
+++ b/bolt_riscv64.go
@@ -1,3 +1,5 @@
+// +build riscv64
+
 package bbolt
 
 // maxMapSize represents the largest mmap size supported by Bolt.
```

--- a/bolt_riscv64.go
+++ b/bolt_riscv64.go
@@ -1,3 +1,5 @@
+// +build riscv64
+
 package bbolt
 
 // maxMapSize represents the largest mmap size supported by Bolt.

